### ISSUE TITLE
[ECS] [request]: Service Connect with Blue/Green deployment

### ISSUE THREAD

**Issue by aetos382** [aws/containers-roadmap#1993]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Please support Service Connect communication between ECS (Fargate) services deployed in CodeDeploy Blue/Green deployment.

**Which service(s) is this request for?**
Fargate, ECS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Service Connect is currently not supported between services using the Blue/Green deployment strategy.

**Are you currently working around this issue?**
Connects ECS services via Application Load Balancer.

**Comment by herrhound** [comment]
@aetos382, thanks for submitting the feature request. Can you tell me more about your deployment configuration and let me know if you use a test listener with your ALB?

**Comment by power-cut** [comment]
Facing the same issue.
Currently using cloudmap, service discovery with route 53 to handle blue green deployment as it fits my use case (for now). App mesh on the other hand does involve certain learning curve.
Since the introduction of service connect, interconnecting services became a breeze but happiness was not for long when we discovered that it doesn't support blue/green deployment by codedeploy.

Apart of 80 and 443 we are using test listeners - port 8000 for our app services

**Comment by michal-maslaniec-tf** [comment]
Hello @vibhav-ag!
When you have a moment, could you please let me know if there's an estimated timeline for when the work might be completed?

**Comment by brainwipe** [comment]
Do we have a timeline for this support?

If not, then you need to update your ECS documentation to make it plainly clear that Service Connect does not support Blue/Green deployments. If, like me, you are working through containerising your application, deployment with Blue/Green comes just after getting Service Connect up and running.

At least you must tell new users what the intended setup should be for Blue/Green. Is it to do everything through "Service Discovery" or will I get that set up only to find Code Deploy doesn't support that either.

**Comment by brainwipe** [comment]
Also, FYI, when you set up Service Discovery then it recommends Service Connect even on services that are blue/green'd. 
![image](https://github.com/user-attachments/assets/7a339870-522c-471d-b8a9-442738c2a189)

**Comment by Harsh4999** [comment]
Is there any update on this we are really stuck in a bad spot we never knew Service Connect is not supported with B/G deployment and now when we are trying to find solutions for interservice communication we cannot use Service Cannot or even VPC Lattice.

**Comment by Yogagh** [comment]
We feel like AWS in-general for reasons like "quick to go to market" strategy, many times they add changes in the platforms like ECS here for ServiceConnect capability but that is not compatible for AWS CodeDeploy Blue/Green deployments. We are working on a very large scale Cloud Modernization project, where we do AWS CodeDeploy Blue/Green deployments and at the same time for near zero latency we would like to leverage this ECS Service's ServiceConnect capabilities so that tens of our ECS services can talk to each other within the ECS platform without taking the ELB route. 

Is there any update on this as we are really stuck in a bad spot we never knew Service Connect is not supported with AWS CodeDeploy's B/G deployment?

**Comment by digglife** [comment]
Now Supported.
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/blue-green-deployment-implementation.html

**Comment by Yogagh** [comment]
@digglife AWS ECS ServiceConnect feature now working with **### _July 15th 2025_** announced AWS ECS platform native Blue/Green deployments enablement. 
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-type-blue-green.html
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/document_history.html

We are looking for AWS ECS ServiceConnect working with AWS ECS CodeDeploy Blue/Green deployments which is not yet supported - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-connect-concepts-deploy.html

**Comment by digglife** [comment]
@Yogagh 
Yes, I'm facing the same problem. Now that they support ECS native B/G deployment I guess CodeDeploy support will never come. I'm now evaluating the migration.

**Comment by vibhav-ag** [comment]
https://aws.amazon.com/blogs/containers/streamline-service-to-service-communication-during-deployments-with-amazon-ecs-service-connect/
